addons/website/models/ir_ui_menu.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, tools
+from odoo.http import request
+
+
+class IrUiMenu(models.Model):
+    _inherit = 'ir.ui.menu'
+
+    @api.model
+    @tools.ormcache_context('self._uid', keys=('lang', 'force_action',))
+    def load_menus_root(self):
```

Comment:
Note that this method is only used from `website` in our codebase.
In master, we may want to move it to `website` and always force the action.
I will open another PR once this is ported and merged in master as it will need to approval or the framework-py.